src/ServiceControl.Transports/AbstractQueueLengthProvider.cs

```diff
@@ -0,0 +1,25 @@
+namespace ServiceControl.Transports
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Extensions.Hosting;
+
+    public abstract class AbstractQueueLengthProvider(TransportSettings settings, Action<QueueLengthEntry[], EndpointToQueueMapping> store)
+        : BackgroundService, IProvideQueueLength
+    {
+        protected Action<QueueLengthEntry[], EndpointToQueueMapping> Store { get; private set; } = store;
+
+        protected string ConnectionString { get; } = settings.ConnectionString;
+
+        public override async Task StartAsync(CancellationToken cancellationToken)
+        {
+            // BackgroundService in .NET awaits the returned task to be completed. Queue providers might be doing
+            // some heavy lifting before yielding so we are forcing things to be offloaded here by default.
+            await Task.Yield();
```

Comment:
This feels strange to me, I feel that should be up to the implementations

Reply:
Yeah it is weird but this is actually an ongoing issue in the .NET runtime repository that MS will eventually fix. Given that all our queue length providers follow the same pattern and this only affects the startup and not really the hot path I figured it is better to put it into a cross-cutting place